Homework5: Ekaterina Zavadskaya

KateZavadskaya opened the pull request:
Add homework5 and minor fix in LICENCE.

KateZavadskaya committed:
Feat(hm5): add homework5

Add 5 files: hm5_1.py - hm5_5.py

KateZavadskaya committed:
Doc(text): add correct text head in LICENCE

Add Ekaterina Zavadskaya in the head of LICENCE

Doc: LICENCE

Stml89 reviewed (changes requested):
Please fix linters

KateZavadskaya committed:
Fix(linter): fix linter error

Add docstring, change variable names

KateZavadskaya committed:
Fix(linter): Improve linter C0103

Improve linter in hm5_1.py - hm5_5.py

Docs: hm5_1.py - hm5_5.py

KateZavadskaya committed:
Fix(Actions): Fix Flake8 and Pycodestyle

Delete extra whitespace

KateZavadskaya commented:
@Stml89 It seems I'm successful 💃

Stml89 commented on hm5_1.py:
```diff
@@ -0,0 +1,29 @@
+
+# Способ 3.
+
+print("www.my_site.com#about".replace("#", "/"))
```
it is the same as first solution

Stml89 commented on hm5_3.py:
```diff
@@ -0,0 +1,15 @@
+
+X = TXT_2[0]
+TXT_2[0] = TXT_2[1]
+TXT_2[1] = X
```
not the best solution. 

>>> TXT = "Ivanou Ivan"
>>> TXT_2 = TXT.split()
>>> TXT = " ".join([TXT_2[1], TXT_2[0]])
>>> print(TXT)
Ivan Ivanou

Stml89 commented on hm5_4.py:
Why do you need a new `any_txt_1` variable?


ANY_TXT = ""
print(f"Исходный текст: '{ANY_TXT}'")

CLEAN_TXT = ANY_TXT.strip()
print(f"Чистый текст: '{CLEAN_TXT}'")

Stml89 commented on hm5_5.py:
The same, too much extra variables

Stml89 reviewed (commented):
Please, read my comments and take them into account in the next HWs. Please fix them before merge

KateZavadskaya commented:
> Please, read my comments and take them into account in the next HWs. Please fix them before merge

Analizing your comment and fixing my mistakes, thank you!👍

KateZavadskaya committed:
Fix(code): fix code mistakes

Analize comments and fix mistakes:
- hm5_1: leave omly 2 variants, delete extra variabls
- hm5_2: delete extra variable
- hm5_3: modify code
- hm5_4: modify code
- hm5_5: delete extra variable

Docs: hm5_1.py - hm5_5.py

KateZavadskaya committed:
Fix(linter): Fix linter mistakes

Fix linter, flake8, pylint mistakes

Docs: hm5_1.py, hm5_3.py

KateZavadskaya committed:
Fix(linter): add one whitespace

Add one whitespace at the end of hm5_1.py

Doc: hm5_1.py

KateZavadskaya commented:
@Stml89 I do my best in resolving your comments. Can you check my results, please.

Stml89 commented on hm5_1.py:
```diff
@@ -0,0 +1,20 @@
+TXT_11 = "www.my_site.com#about"
+
+LST = list(TXT_11)
+X = LST.index("#")
```
why you can not use simple 'find' method?
https://www.w3schools.com/python/ref_string_find.asp

KateZavadskaya replied:
Understand your message.

KateZavadskaya replied:
Will redo, ok 👌

Stml89 commented on hm5_3.py:
```diff
@@ -0,0 +1,7 @@
+
+TXT = "Ivanou Ivan"
+
+print(TXT.split()[1], TXT.split()[0])
```
here you execute method split() twice, not reaaly good practice 
Better:
something = TXT.split()
print(something[1], something[0])

KateZavadskaya replied:
I see, will take a note!

KateZavadskaya replied:
Will redo, ok 👌

KateZavadskaya replied:
Will redo, ok 👌

KateZavadskaya committed:
Fix(code): add code improvements

Add code improvements:
-hm5_1.py: delete 'split'method, add 'find'method
-hm5_3.py: add variable, delete extra 'split'method

Docs: hm5_1.py, hm5_3.py

KateZavadskaya commented:
@Stml89 fixed your comments, check it 🤞, please.
